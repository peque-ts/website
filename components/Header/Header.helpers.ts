interface NavItem {
  name: string;
  link: string;
}

export const NAV_ITEMS: NavItem[] = [
  {
    name: 'Docs',
    link: '/docs',
  },
  {
    name: 'Enterprise',
    link: '/enterprise',
  },
  {
    name: 'Sponsors',
    link: '/sponsors',
  },
];
