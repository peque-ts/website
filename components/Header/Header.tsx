import clsx from 'clsx';
import Link from 'next/link';
import React from 'react';

import { Container } from '../Container';
import { Icon } from '../Icon';
import { MainNav } from '../MainNav';

interface Props {
  text?: string;
  bgClassName?: string;
}

export const Header: React.VFC<Props> = ({ text, bgClassName }) => (
  <header className={clsx('fixed top-0 z-10 h-16 w-full', bgClassName)}>
    <Container className="flex h-full items-center justify-between">
      <div className="flex items-center">
        <Link href="/">
          <a className="flex">
            <Icon name="Logo" size={40} />
          </a>
        </Link>
        <h4>{text}</h4>
      </div>
      <div className="flex items-center space-x-8">
        <MainNav bgClassName={bgClassName} />
        <Link href="https://github.com/pequehq">
          <a className="flex">
            <Icon name="GitHub" size={24} />
          </a>
        </Link>
      </div>
    </Container>
  </header>
);
