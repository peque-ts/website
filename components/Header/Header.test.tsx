import { render, screen } from '@testing-library/react';

import { Header } from './Header';

describe('Header', () => {
  it('should render', () => {
    const { asFragment } = render(<Header />);

    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
