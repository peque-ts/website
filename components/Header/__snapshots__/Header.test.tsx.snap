// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<DocumentFragment>
  <div
    class="h-16 bg-primary-600"
  >
    <div
      class="container h-full mx-auto px-2 flex items-center justify-between"
    >
      <a
        href="/"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </a>
      <div
        class="flex items-center space-x-8"
      >
        <nav>
          <ul
            class="flex space-x-8"
          >
            <li>
              <a
                class="text-lg"
                href="/docs"
              >
                Docs
              </a>
            </li>
            <li>
              <a
                class="text-lg"
                href="/enterprise"
              >
                Enterprise
              </a>
            </li>
            <li>
              <a
                class="text-lg"
                href="/sponsors"
              >
                Sponsors
              </a>
            </li>
          </ul>
        </nav>
        <a
          href="https://github.com/peque-ts"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Peque on GitHub"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
